TWLight/users/signals.py

```diff
@@ -1,4 +1,41 @@
-import django.dispatch
+from datetime import timedelta
+from django.dispatch import receiver, Signal
+from django.db.models.signals import post_save
+from TWLight.users.models import Authorization
+from .models import Partner, Stream
+
 
 class Notice(object):
-    user_renewal_notice = django.dispatch.Signal(providing_args=['user_wp_username', 'user_email', 'user_lang', 'partner_name', 'partner_link'])
+    user_renewal_notice = Signal(providing_args=['user_wp_username', 'user_email', 'user_lang', 'partner_name', 'partner_link'])
+
+
+
+@receiver(post_save, sender=Partner)
+@receiver(post_save, sender=Stream)
+def update_partner_authorization_expiry(sender, instance, **kwargs):
+    """
+    Updates missing expiration dates upon resource updates.
+    Mostly cribbed from
+    TWLight.applications.models.post_receive_commit
+    and
+    TWLight.users.management.commands.authorization_backfill
+    Could factor out the common code.
+    """
+
+    print(sender)
+    if sender == Partner:
+        partner = instance
+    elif sender == Stream:
+        partner = instance.partner
+
+    if partner.account_length or partner.requested_access_duration:
```

Comment:
`requested_access_duration` is effectively a dummy field for generating the applications in a consistent way. We should probably just check for `.PROXY` to be clearer about what's happening here?

Reply:
not sure what I was thinking here